fe/fe-core/src/main/java/com/starrocks/external/HiveMetaStoreTableUtils.java

```diff
@@ -103,4 +110,82 @@ public static long getPartitionStatsRowCount(HiveMetaStoreTableInfo hmsTable, Li
         }
         return numRows;
     }
+
+    public static Map<String, FieldSchema> getAllHiveColumns(Table table) {
+        List<FieldSchema> unPartHiveColumns = table.getSd().getCols();
+        List<FieldSchema> partHiveColumns = table.getPartitionKeys();
+        Map<String, FieldSchema> allHiveColumns = unPartHiveColumns.stream()
+                .collect(Collectors.toMap(FieldSchema::getName, fieldSchema -> fieldSchema));
+        for (FieldSchema hiveColumn : partHiveColumns) {
+            allHiveColumns.put(hiveColumn.getName(), hiveColumn);
+        }
+        return allHiveColumns;
+    }
+
+    public static List<FieldSchema> getAllColumns(Table table) {
+        List<FieldSchema> allColumns = table.getSd().getCols();
+        List<FieldSchema> partHiveColumns = table.getPartitionKeys();
+        allColumns.addAll(partHiveColumns);
+        return allColumns;
+    }
+
+    public static Type convertColumnType(String hiveType) throws DdlException {
+        String typeUpperCase = Utils.getTypeKeyword(hiveType).toUpperCase();
+        PrimitiveType primitiveType;
+        switch (typeUpperCase) {
+            case "TINYINT":
+                primitiveType = PrimitiveType.TINYINT;
+                break;
+            case "SMALLINT":
+                primitiveType = PrimitiveType.SMALLINT;
+                break;
+            case "INT":
+            case "INTEGER":
+                primitiveType = PrimitiveType.INT;
+                break;
+            case "BIGINT":
+                primitiveType = PrimitiveType.BIGINT;
+                break;
+            case "FLOAT":
+                primitiveType = PrimitiveType.FLOAT;
+                break;
+            case "DOUBLE":
+            case "DOUBLE PRECISION":
+                primitiveType = PrimitiveType.DOUBLE;
+                break;
+            case "DECIMAL":
+            case "NUMERIC":
+                primitiveType = PrimitiveType.DECIMAL32;
+                break;
+            case "TIMESTAMP":
+                primitiveType = PrimitiveType.DATETIME;
+                break;
+            case "DATE":
+                primitiveType = PrimitiveType.DATE;
+                break;
+            case "STRING":
```

Comment:
you may add binary type. user behavior of version 2.2 should be consistent.

Reply:
> you may add binary type. user behavior of version 2.2 should be consistent.

```
            case "STRING":
            case "BINARY":
                return ScalarType.createDefaultString();
```